ios-base/ViewModel/Onboarding/SignUpViewModel.swift

```diff
@@ -55,13 +55,19 @@ class SignUpViewModelWithEmail {
   
   func signup() {
     state = .loading
-    UserAPI.signup(email, password: password,
+    UserService.sharedInstance.signup(email, password: password,
                    avatar64: UIImage.random(),
-                   success: { [weak self] _ in
+                   success: { [weak self] in
                     self?.state = .signedUp
                    },
                    failure: { [weak self] error in
-                    self?.state = .error(error.localizedDescription)
+                    if let apiError = error as? APIError,
```

Comment:
Maybe all this logic can be moved to `APIError`, then consumers would only do something like `apiError.mainError`. WDYT?

Reply:
I like it, i'll add some helpers